streaming/src/main/scala/org/apache/spark/streaming/scheduler/JobScheduler.scala

```diff
@@ -193,6 +197,14 @@ class JobScheduler(val ssc: StreamingContext) extends Logging {
     listenerBus.post(StreamingListenerOutputOperationCompleted(job.toOutputOperationInfo))
     logInfo("Finished job " + job.id + " from job set of time " + jobSet.time)
     if (jobSet.hasCompleted) {
+      // submit fake BatchCompleted event to show missing inputInfo on Streaming UI
+      inputInfoMissedTimes.foreach (time => {
+        val streamIdToInputInfos = inputInfoTracker.getInfo(time)
+        val fakeJobSet = JobSet(time, Seq(), streamIdToInputInfos)
+        listenerBus.post(StreamingListenerBatchCompleted(fakeJobSet.toBatchInfo))
```

Comment:
Will this change the semantics if posing fake `StreamingListenerBatchCompleted`, if user's code rely on this, will this break their assumptions?